rllib/core/models/tests/test_catalog.py

```diff
@@ -256,6 +263,91 @@ def test_get_dist_cls_from_action_space(self):
                 actions = dist.sample()
                 assert action_space.contains(actions.numpy()[0])
 
+    def test_catalog_and_algorithm_config(self):
+        """Test if we can pass catalog to algorithm config and it ends up inside
+        RLModule and is used to build models there."""
+
+        class MyCatalog(PPOCatalog):
+            pass
+
+        config = (
+            PPOConfig()
+            .rl_module(
+                rl_module_spec=SingleAgentRLModuleSpec(
+                    module_class=PPOTorchRLModule, catalog_class=MyCatalog
```

Comment:
It should work so that i don't have to specify module_class=PPOTorchRLModule if I am not overriding the default.

Reply:
I introduced a couple of modifications to AlgorithmConfig, the MARL and SARL specs that deal with this.
Basically, we need some update() mechanism that gracefully overwrites defaults with user-set data.